components/slack/src/component.ts

```diff
@@ -0,0 +1,46 @@
+import { ILoggerComponent } from '@well-known-components/interfaces'
+import { WebClient } from '@slack/web-api'
+import { SlackConfig, SlackMessage, ISlackComponent } from './types'
+
+export function createSlackComponent(
+  components: Pick<{ logs: ILoggerComponent }, 'logs'>,
+  config: SlackConfig
+): ISlackComponent {
+  const { logs } = components
+  const logger = logs.getLogger('slack')
+  let token: string | undefined = config.token
+
+  if (!token) {
+    logger.error('No token provided')
+    throw new Error('No token provided')
+  }
+
+  const client = new WebClient(token)
+
+  async function sendMessage(message: SlackMessage): Promise<void> {
+    try {
+      if (!message.channel) {
```

Comment:
Channel is always defined in the `SlackMessage` type, would you mind removing this check?